### PR TITLE
Added a few device GUIDs to `is_xinput_device` fixing controller problems

### LINKED ISSUE #76724: Xbox bluetooth controller detected as 2 devices, triggering 2 events instead of 1
### Godot version

3.5.2 stable, 3.6 beta 1

### System information

Windows 11, all amd system, GLES3

### Issue description

Printed output says that the controller is in fact two controllers:
```
XInput Gamepad
Bluetooth LE XINPUT compatible input device
```
controllers of other makes only report as a single device (tested with nintendo pro controller, dual sense, steam deck)

Tested controllers:
- xbox elite controller
- xbox one s
- ~~haven't tested with xbox series controller, will report back when i have~~ Xbox series *(pink)* works fine

Effect:
- on a fresh new project it seems that the issue is only that the gamepad is reported as two
- in a project we are working on in godot 3.5.2, the double sent events propagate into the game (skips a button when going up for instance)

I am aware of https://github.com/godotengine/godot/issues/41684 and https://github.com/godotengine/godot/pull/71784 which mentions that a fix had been cherry picked, in my testing however the issue remains.

### Steps to reproduce

- create a simple ui scene with a couple of buttons
- add a script that prints the device name using: `print(Input.get_joy_name(event.device))` when event is InputEventJoypadButton in the _input function
- run the game and press a button on a bluetooth connected xbox gamepad

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by mrTag**
The Xbox Elite Wireless gamepad has strange input problems in Windows, which we identified as the same problem as in this PR: https://github.com/godotengine/godot/pull/71784 and fixed in the same way (by adding another hardcoded device ID to JoypadWindows::is_xinput_device).

The fix was tested with a Xbox Elite Wireless gamepad and it works.

Update: added more GUIDs from SDL, as discussed below.
Probably also fixes https://github.com/godotengine/godot/issues/76724 now.

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks for the patch! It seems correct and should be fine to merge.

There are more issues with this code that we'll need to address eventually though. Both #71784 and this PR added missing devices, but there are quite a few more Xbox Bluetooth/BLE controllers with a similar issue.

This code was originally based off SDL's, but it seems like they changed their approach in 2019 with https://github.com/libsdl-org/SDL/commit/650964461ee9202b8f073bc8c3369530fab1a1a4 

We can see that back then they had a few more devices which we still lack:
```
    static GUID IID_ValveStreamingGamepad = { MAKELONG(0x28DE, 0x11FF), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
    static GUID IID_X360WiredGamepad = { MAKELONG(0x045E, 0x02A1), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
    static GUID IID_X360WirelessGamepad = { MAKELONG(0x045E, 0x028E), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneWiredGamepad = { MAKELONG(0x045E, 0x02FF), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneWirelessGamepad = { MAKELONG(0x045E, 0x02DD), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneNewWirelessGamepad = { MAKELONG(0x045E, 0x02D1), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneSWirelessGamepad = { MAKELONG(0x045E, 0x02EA), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneSBluetoothGamepad = { MAKELONG(0x045E, 0x02E0), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	// Missing in Godot.
    static GUID IID_XOneEliteWirelessGamepad = { MAKELONG(0x045E, 0x02E3), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
```

And now we have a couple entries which they didn't have back in 2019:
```
	static GUID IID_XSWirelessGamepad = { MAKELONG(0x045E, 0x0B13), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
	static GUID IID_XEliteWirelessGamepad = { MAKELONG(0x045E, 0x0B05), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
```

Longer term, we want to investigate leveraging SDL2 directly for gamepad mapping instead of trying to play catch with them and vendors.

Short term, I guess adding the other missing ones identified above would be useful. Let me know if you're interested in doing it, or if I should pick it up.

File: platform/windows/joypad_windows.cpp [hunk 1]
```diff
@@ -102,11 +102,13 @@ bool JoypadWindows::is_xinput_device(const GUID *p_guid) {
 	static GUID IID_X360WiredGamepad = { MAKELONG(0x045E, 0x02A1), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
 	static GUID IID_X360WirelessGamepad = { MAKELONG(0x045E, 0x028E), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
 	static GUID IID_XSWirelessGamepad = { MAKELONG(0x045E, 0x0B13), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
+	static GUID IID_XEliteWirelessGamepad = { MAKELONG(0x045E, 0x0B05), 0x0000, 0x0000, { 0x00, 0x00, 0x50, 0x49, 0x44, 0x56, 0x49, 0x44 } };
 
 	if (memcmp(p_guid, &IID_ValveStreamingGamepad, sizeof(*p_guid)) == 0 ||
 			memcmp(p_guid, &IID_X360WiredGamepad, sizeof(*p_guid)) == 0 ||
 			memcmp(p_guid, &IID_X360WirelessGamepad, sizeof(*p_guid)) == 0 ||
-			memcmp(p_guid, &IID_XSWirelessGamepad, sizeof(*p_guid)) == 0)
+			memcmp(p_guid, &IID_XSWirelessGamepad, sizeof(*p_guid)) == 0  ||
```
[the remark this answers is not in the document]
Comment (mrTag) [reply platform/windows/joypad_windows.cpp]: Ah, right, thanks! Need to pay more attention to clang tidy 😅 Changed and force pushed.

**Comment by mrTag** [comment]
Thanks for the list @akien-mga! Absolutely makes sense to include those in godot as well until the problem is solved for good. I can quickly do that, should I just change this PR or open a new one?

**Comment by akien-mga** [comment]
It's best if you can amend this PR directly to add all the gamepads we identified so far at once.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! Technically your first merged contribution as far as GitHub is concerned, but I know you've done a ton to help fix issues on threaded resource loading already. Still, congrats for your first merged PR ;)

**Comment by akien-mga** [comment]
Cherry-picked for 3.6.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.3.

**Pull request merged**